src/Microsoft.Health.Dicom.Core/Extensions/ValidationErrorCodeExtensions.cs

```diff
@@ -0,0 +1,52 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using Microsoft.Health.Dicom.Core.Features.Validation;
+using static Microsoft.Health.Dicom.Core.Features.Validation.ValidationErrorCode;
+
+namespace Microsoft.Health.Dicom.Core.Extensions
+{
+    internal static class ValidationErrorCodeExtensions
+    {
+        private static readonly IReadOnlyDictionary<ValidationErrorCode, string> MessageMap = new Dictionary<ValidationErrorCode, string>()
```

Comment:
should we just use a `switch` statement or expression in the method?

Reply:
There are a bunch of error codes, could add more in the future. I feel a long switch may not be a good idea.

BTW, find something for `switch` performance https://stackoverflow.com/questions/44905/c-sharp-switch-statement-limitations-why,  looks like C# compiler could optimize `switch` with jump table or binary search tree when it's going to big.

Reply:
Ah yes. You can see the optimizations yourself by creating different sizes of switch statements and viewing them in ILSpy!